src/yaml/aslYamlLanguageService.ts

```diff
@@ -0,0 +1,228 @@
+    ): Promise<CompletionList> {
+        const yamldoc = parseYAML(document.getText())
+        const offset = document.offsetAt(position)
+        let currentDoc = matchOffsetToDocument(offset, yamldoc)
+        let atSpace = false
+
+        // if a yaml doc is null, it must be given text to allow auto-completion
+        if (!currentDoc) {
+            currentDoc = initializeDocument(document, offset)
+            // move cursor position into new empty object
+            position.character += 1
+        }
+
+        const yamlCompletions = await completer.doComplete(document, position, false)
+
+        if (!currentDoc) {
+            return { items: [], isIncomplete: false }
+        }
+        // adjust position for completion
+        const node = currentDoc.getNodeFromOffsetEndInclusive(offset)
+        if (node.type === 'array') {
+            // Resolves issue with array item insert text being off by a space
+            position.character -= 1
+        } else if (document.getText().substring(offset, offset + 1) === '"') {
+            // When attempting to auto-complete from inside an empty string, the position must be adjusted within bounds
+            position.character += 1
+        } else if (document.getText().substring(offset - 2, offset) === ': ') {
+            // yaml doesn't allow auto-completion from white-space after certain nodes
+            atSpace = true
+            // initialize an empty string and adjust position to within the string before parsing yaml to json
+            const newText = `${document.getText().substring(0, offset)}""\n${document.getText().substr(offset)}`
+            const parsedDoc = parseYAML(newText)
+            currentDoc = matchOffsetToDocument(offset, parsedDoc)
+            position.character += 1
+        } else if (node.type === 'property' || (node.type === 'object' && position.character !== 0)) {
+            // allow auto-completion of States field from empty space
+            if (document.getText().substring(offset - 2, offset) === '  ') {
+                atSpace = true
+                currentDoc = initializeDocument(document, offset)
+                // move cursor position into new empty object
+                position.character += 1
+            } else {
+                // adjust cursor back after parsing to keep it within States node
+                position.character -= 1
+            }
+        }
+        if (currentDoc) {
+            const aslCompletions : CompletionList  = doCompleteAsl(document, position, currentDoc, yamlCompletions, {
+                ignoreColonOffset: true,
+            })
+
+            aslCompletions.items.forEach(completion => {
+                // format json completions for yaml
+                if (completion.textEdit) {
+                    // textEdit can't be done on white-space so insert text is used instead
+                    if (atSpace) {
+                        completion.insertText = completion.textEdit.newText
```

Comment:
Do we intend to have both of these lines? Seems like duplicate code.

Reply:
Looks like it, removing the duplicate part.